diff.js

```diff
@@ -15,6 +19,35 @@ module.exports = function(config, done) {
 
     var discrepancies = 0;
 
+    var fileName = config.backup.prefix + +new Date() + '.json';
+    var outputFile = fs.createWriteStream(fileName);
+    var writeFile = new stream.Transform({ objectMode: true });
+
+    writeFile._transform = function(record, enc, callback) {
+        outputFile.write(JSON.stringify(record) + '\n');
+        this.push(record);
+        callback();
+    };
+
+    writeFile._flush = function(callback) {
+      outputFile.on('finish', callback);
+      outputFile.end();
+    };
+
+    writeFile.on('end', function() {
+        s3.putObject({
+            Bucket: config.backup.bucket,
+            Key: config.backup.prefix + '/' + fileName,
```

Comment:
What caught my eye here is that `fileName` already contains `config.backup.prefix`